### ISSUE TITLE
Some doubts for total loss?

### ISSUE THREAD

**Issue by wangyanhao0517** [seanie12/SWEP#4]
Sorry to bother you. I found that your code is different from your paper in the loss calculation process. Why is the final loss NLL + beta * KL? Your paper says KL is a part of noise

**Comment by seanie12** [comment]
<img width="297" alt="image" src="https://user-images.githubusercontent.com/19561061/165886163-e9a3bb15-60c4-4a1d-ad41-81436964bc70.png">
Gradient of KL with respect to \theta (the parameter of BERT) is zero since we do not backpropagate it to BERT (torch.no_grad() in [line 42](https://github.com/seanie12/SWEP/blob/main/models.py#L43). 

So the implementation is  consistent with our equation.

**Comment by wangyanhao0517** [comment]
> <img alt="image" width="297" src="https://user-images.githubusercontent.com/19561061/165886163-e9a3bb15-60c4-4a1d-ad41-81436964bc70.png">
> 
> Gradient of KL with respect to \theta (the parameter of BERT) is zero since we do not backpropagate it to BERT (torch.no_grad() in [line 42](https://github.com/seanie12/SWEP/blob/main/models.py#L43).
> So the implementation is consistent with our equation.

**Comment by wangyanhao0517** [comment]
> <img alt="image" width="297" src="https://user-images.githubusercontent.com/19561061/165886163-e9a3bb15-60c4-4a1d-ad41-81436964bc70.png">
> 
> Gradient of KL with respect to \theta (the parameter of BERT) is zero since we do not backpropagate it to BERT (torch.no_grad() in [line 42](https://github.com/seanie12/SWEP/blob/main/models.py#L43).
> So the implementation is consistent with our equation.
Thanks for your reply !
